### PR TITLE
Accept a list of imports allowing non-builtin data types

### PR DESCRIPTION
**Pull request by ashanbrown**
Also allow listing the config directly on the command-line instead of using `-type` arg.  Allows the following syntax:

```
//go:generate go-options -imports=time,url=net/url config
type config struct {
	myURL url.URL
	myDuration time.Duration
}
```

### CODE REVIEW

File: test/sample_test.go [hunk 1]
```diff
@@ -38,57 +40,74 @@ var _ = Describe("Generating options", func() {
 			OptionMyFloat(4.56),
 			OptionMyString("my-string"),
 		)
-		Ω(err).NotTo(HaveOccurred())
-		Ω(cfg.myInt).To(Equal(123))
-		Ω(cfg.myFloat).To(Equal(4.56))
-		Ω(cfg.myString).To(Equal("my-string"))
+		Ω(err).ShouldNot(HaveOccurred())
```
Comment (ashanbrown) [inline test/sample_test.go]: Just some cleanup of ginkgo matchers: Ω and Should should pair up (as opposed to the alternate pairing of Expect and To).

File: main.go [hunk 2]
```diff
@@ -92,7 +116,14 @@ func writeOptionsFile(packageName string, node ast.Node) (found bool) {
...
 
-		if typeSpec.Name.String() != typeName {
+		var typeName string
+		for _, n := range types {
+			if typeSpec.Name.String() == n {
+				typeName = n
+				break
+			}
+		}
+		if typeName == "" {
```
Comment (jstncbllr) [inline main.go]: i sure do hope the "go contracts" design gets implemented and gives up a standard way to filter

File: main.go [hunk 3]
```diff
@@ -137,6 +172,20 @@ func writeOptionsFile(packageName string, node ast.Node) (found bool) {
...
 		}
 
+		var importList []Import
+		if imports != "" {
+			for _, s := range strings.Split(imports, ",") {
+				parts := strings.Split(s, "=")
+				if len(parts) == 1 {
+					importList = append(importList, Import{Path: parts[0]})
+				} else if len(parts) == 2 {
+					importList = append(importList, Import{Alias: parts[0], Path: parts[1]})
```
Comment (jstncbllr) [inline main.go]: consider checking if the second part ends with "/" + first part?  If so we wouldn't need to write out the alias?  are are you just assuming that go fmt will be run?

File: test/sample_test.go (hunk 1 above)
Comment (jstncbllr) [reply test/sample_test.go]: one of the ways ginkgo api isn't super tight.  but, still a fan of course.

File: main.go (hunk 3 above)
Comment (ashanbrown) [reply main.go]: We do run gofmt by default.  I think it's ok just to leave it up to the user if they want an alias or not.

**Pull request merged**